08_Langy/parser.y

```diff
@@ -229,3 +236,47 @@ void genStatementList( FILE* fout
     curstmt = curstmt->next;
   }
 }
+
+
+void freeStmtNode( struct stmtnode* stmt)
+{
+  if( stmt != NULL)
+  {
+    freeStmtNode( stmt->next);
+    freeStmtNode( stmt->body);
+    freeExprNode( stmt->expr);
+    
+    free(stmt);
+    stmt = NULL;

```

Comment:
This needs to be in the calling function -- struct stmtnode* is not a reference parameter. Ditto for the other free functions.